src/main.rs

```diff
@@ -70,7 +71,13 @@ async fn main() {
 
     let addr = SocketAddr::new(IpAddr::V4(Ipv4Addr::UNSPECIFIED), port);
 
-    let mut engine = Engine::new(client.clone(), logger.new(o!()));
+    let mut managed_index = ManagedIndex::new(Duration::from_secs(20));
+    let index = managed_index.index();
+    tokio::spawn(async move {
+        managed_index.refresh_at_interval().await;
+    });
```

Comment:
pretty sure thats redundant

```suggestion
    tokio::spawn(managed_index.refresh_at_interval());
```

Reply:
It doesn't work because it's not taking ownership of `managed_index`.